airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -113,6 +113,16 @@ public ConnectionManagerWorkflowImpl() {}
   public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws RetryableException {
     try {
       try {
+        connectionId = connectionUpdaterInput.getConnectionId();
+
+        final Duration timeToWait = getTimeToWait(connectionId);
+        Workflow.await(timeToWait, () -> skipScheduling() || connectionUpdaterInput.isFromFailure());
+
+        // Clean the job state by failing any jobs for this connection that are currently non-terminal.
+        // This catches cases where the temporal workflow was terminated and restarted while a job was
+        // actively running, leaving that job in an orphaned and non-terminal state.
+        ensureCleanJobState(connectionUpdaterInput.getJobId());
```

Comment:
I believe we explicitly want the `ensureCleanJobState()` to be called _before_ the `Workflow.await()`. 

If a connection is in a bad state (e.g. say the temporal workflow was terminated while a sync job was running), then a manual operation (e.g. cancelling that "running" job) will cause the workflow to be restarted. If `ensureCleanJobState()` is not executed until after the `Workflow.await()`, then when the manual operation is performed, nothing in the UI will be updated until the scheduling time has passed since the start time of that running job, at which point it will be set to failed and a new job will be immediately kicked off.

If `ensureCleanJobState()` happens at the beginning of the connection manager workflow, as is implemented today, then the user gets instant feedback of the "running" job being set to failed when they perform a manual operation like a cancelled, which seems like a better UX to me. Plus, that approach feels slightly better in terms of code organization, since it is basically saying that the connection manager workflow should start off from a clean state, rather than waiting to clean up the state until it is ready to launch a new job.